bugbug/utils.py

```diff
@@ -468,3 +470,20 @@ def should_skip(changeset):
 
 def get_physical_cpu_count() -> int:
     return psutil.cpu_count(logical=False)
+
+
+def extract_metadata(body) -> dict:
+    match_list = re.findall(r"<!--\s@(\w+):\s(.+)\s-->", body)
+    metadata_dict = dict(match_list)
+    return metadata_dict
+
+
+def extract_private(issue_body) -> tuple:
+    private_url = extract_metadata(issue_body).get("private_url", "").strip()
+    private_issue_path = urllib.parse.urlparse(private_url).path
+
+    if private_issue_path:
+        owner, repo, _, number = tuple(private_issue_path.split("/")[1:])
+        return owner, repo, number
+
+    return ()
```

Comment:
Nit: `return None`, I prefer to use None when a value is to be considered as missing (like in this case). Then in the code using the function, you can do `if XXX is None` instead of `if not XXX` to be more precise